src/components/menu/menu.style.js

```diff
@@ -11,6 +11,7 @@ const StyledMenuWrapper = styled(Box).attrs({ as: "ul" })`
   list-style: none;
   margin: 0;
   padding: 0;
+  z-index: ${({ theme }) => theme.zIndex.popover};
```

Comment:
I think you should be adding this to the submenu, not the whole `Menu`. If you look in `src/menu/__internal__/submenu/submenu.style.js` you'll see `z-index: 10` being applied to the `StyledSubmenu` - I think that's what you should be changing here